### PR TITLE
Remove redundant voting and proposal procedures functions. Update cardano-api.

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Remove redundant voting and proposal procedures functions.
    Update cardano-api-9.2.0.0.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
   - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
   - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

This PR is stacked on top of: 
- https://github.com/IntersectMBO/cardano-cli/pull/865

This PR should be reviewed on commit-by-commit basis.

This PR introduces additional minor refactoring:
- Replace usage of `L.Coin` with `Lovelace` type alias and a pattern synonym
- Replace usages of `ListMap.toList` and   `ListMap.fromList` with `IsList(toList, fromList)`.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs [hunk 1]
```diff
@@ -1104,16 +1104,17 @@ writePoolState mOutFile serialisedCurrentEpochState = do
     pure (decodePoolState serialisedCurrentEpochState)
       & onLeft (left . QueryCmdPoolStateDecodeError)
 
-  let hks =
+  let hks :: [L.KeyHash L.StakePool StandardCrypto]
+      hks =
         toList $
-          Set.fromList $
+          fromList @(Set (L.KeyHash L.StakePool StandardCrypto)) $
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs]: Is the type annotation needed? In this case, I'm not sure this is really an improvement, but fine nevertheless.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs [hunk 2]
```diff
@@ -198,7 +199,7 @@ runTransactionBuildCmd
           <$> readTxGovernanceActions eon proposalFiles
 
     -- the same collateral input can be used for several plutus scripts
-    let filteredTxinsc = toList $ Set.fromList txinsc
+    let filteredTxinsc = toList @(Set _) $ fromList txinsc
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs]: Same here, I prefer keeping `Set.fromList`. It's more KISS.

Non-blocking, leave it like this if you want (since this is more consistent).

File: cardano-cli/src/Cardano/CLI/Json/Friendly.hs [hunk 3]
```diff
@@ -2,9 +2,13 @@
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE PatternSynonyms #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeOperators #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/Json/Friendly.hs]: Why is this required?

File: cardano-cli/src/Cardano/CLI/Json/Friendly.hs [hunk 4]
```diff
@@ -247,11 +247,12 @@ friendlyTxBodyImpl
...
-                          Nothing -> []
-                          Just (Featured _ TxProposalProceduresNone) -> []
-                          Just (Featured _ (TxProposalProcedures lProposals _witnesses)) ->
-                            ["governance actions" .= (friendlyLedgerProposals cOnwards $ toList lProposals)]
+                        conwayEraOnwardsConstraints cOnwards $
+                          case txProposalProcedures of
+                            Nothing -> []
+                            Just (Featured _ pp) -> do
+                              let lProposals = toList $ convProposalProcedures pp
+                              ["governance actions" .= (friendlyLedgerProposals cOnwards lProposals)]
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/Json/Friendly.hs:255]: Previously, in the case `TxProposalProceduresNone`, the output JSON would not contain the key `governance actions`.

Not we will output `"governance actions" = []`. Is this intended?

**Review by smelc — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Json/Friendly.hs (hunk 4 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/Json/Friendly.hs:255]: Good catch. That shouldn't happen. I'll fix this.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs]: Good catch. I thought we're going through set here to deduplicate elements - but the map keys are already unique, so it's pointless. I'll remove the conversion.

Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs]: This can be simplified, I'll update it.

File: cardano-cli/src/Cardano/CLI/Json/Friendly.hs (hunk 3 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/Json/Friendly.hs]: Not sure to be honest. HLS inserts this when you're using `{-# HLINT ignore ...` pragma. Otherwise I'm getting a warning.

Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/Json/Friendly.hs]: Probably because the pragma is above `module ...` ?

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs]: Good catch. I've replaced it with `nubOrd` which has the same asymptotics for unsorted lists.

File: cardano-cli/src/Cardano/CLI/Types/Common.hs [hunk 5]
```diff
@@ -639,3 +641,9 @@ data ParserFileDirection
   = Input
   | Output
   deriving (Eq, Show)
+
+-- TODO move to cardano-api
+pattern Lovelace :: Integer -> Lovelace
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Types/Common.hs]: Let's just use `Coin`.

**Review by Jimbo4350 — APPROVED** [review]
LGTM but let's just use the `Coin` constructor.
